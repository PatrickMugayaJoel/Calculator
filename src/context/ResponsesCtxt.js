import React, {useState, createContext} from 'react';

export const ResponseContext = createContext();

export const ResponseProvider = props => {
    const [responces, setResponces] = useState([]);
    return (
        <ResponseContext.Provider value={[responces, setResponces]}>
            {props.children}
        </ResponseContext.Provider>
    );
}
