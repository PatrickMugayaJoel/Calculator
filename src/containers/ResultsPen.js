import React, { useContext } from 'react';
import TableRow from "../components/TableRow";
import {ResponseContext} from "../context/ResponsesCtxt";

export const ResultPen = () => {
    const [responces] = useContext(ResponseContext);
    return (
        <div className="col-sm-8">
        <div className="results-pannel full-height">
          <span className="title">Results</span>
          <div className="table">
            <table>
              <thead>
                <tr>
                  <th>Number&nbsp;1</th>
                  <th>Number&nbsp;2</th>
                  <th>Response</th>
                  <th>Expected</th>
                  <th>Passed</th>
                  <th>Remove</th>
                </tr>
              </thead>
              <tbody>
                {  
                  responces.map(
                      (result, index) => <TableRow key={index} id={index} result={result} />
                    )
                }
              </tbody>
            </table>
          </div>
        </div>
    </div>
    );
}

export default ResultPen;