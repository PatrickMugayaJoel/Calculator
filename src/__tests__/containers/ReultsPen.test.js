import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import ResultsPen from '../../containers/ResultsPen';
import {ResponseProvider} from "../../context/ResponsesCtxt";

describe('ResultsPen container snapshot', () => {
    let wrapper = shallow(
        <ResponseProvider>
          <ResultsPen />
        </ResponseProvider>
      );
  
    it('matches snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
