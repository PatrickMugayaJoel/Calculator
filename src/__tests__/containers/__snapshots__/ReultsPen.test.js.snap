// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsPen container snapshot matches snapshot 1`] = `
<ContextProvider
  value={
    Array [
      Array [],
      [Function],
    ]
  }
>
  <ResultPen />
</ContextProvider>
`;
